### ISSUE TITLE
Number Input Buttons need Descriptive Labels and Input Needs aria-live

### ISSUE THREAD

**Issue by emyarod** [carbon-design-system/carbon#1743]
copied from https://github.com/IBM/carbon-components-react/issues/1658

---

## Detailed description

The [latest version of number input](https://deploy-preview-1657--carbon-components-react.netlify.com/?selectedKind=NumberInput&selectedStory=Default&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Fstories%2Fstories-panel) has two buttons that when clicked increment the number box. The plus button should be labeled "+ increment" and the - button should be labeled "- Decrement". The input also needs to have aria-live="polite" and aria-atomic="true" so that the screen reader always reads the whole number in the input box rather than just the numbers React changes.
As the buttons are currently, they both have the same label. Here are the button elements:

	<button class="bx--number__control-btn up-icon" type="button"><svg class="up-icon" fill-rule="evenodd" height="5" role="img" viewBox="0 0 10 5" width="10" aria-label="choose a number" alt="choose a number"><title>choose a number</title><path d="M0 5L5 .002 10 5z"></path></svg></button>
	<button class="bx--number__control-btn down-icon" type="button"><svg class="down-icon" fill-rule="evenodd" height="5" role="img" viewBox="0 0 10 5" width="10" aria-label="choose a number" alt="choose a number"><title>choose a number</title><path d="M0 0l5 4.998L10 0z"></path></svg></button>

## Steps to reproduce the issue

1. Using NVDA visit the [components page](https://deploy-preview-1657--carbon-components-react.netlify.com/?selectedKind=NumberInput&selectedStory=Default&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Fstories%2Fstories-panel)
2. hit m to jump to the frame
3. NVDA announces: "Button choose a number graphic choose a number". This is the up arrow.
4. Hit down arrow
5. NVDA announces "Button choose a number graphic choose a number". This is the down arrow.
6. hit enter on one of the buttons and NVDA says nothing.

## Expected Behavior

NVDA should say an informative text on the button and when one hits enter NVDA should say what is changed.
The svg does not need any role, alt, or aria-label. The text that should change is in the title element. (I need to test to make sure this is correct, but text should not be duplicated).
